src/Compilers/CSharp/Portable/Binder/Semantics/OverloadResolution/OverloadResolutionResult.cs

```diff
@@ -970,6 +973,7 @@ private static bool HadLambdaConversionError(DiagnosticBag diagnostics, Analyzed
             {
                 Debug.Assert(argument.Kind != BoundKind.OutDeconstructVarPendingInference);
                 Debug.Assert(argument.Kind != BoundKind.OutVariablePendingInference);
+                Debug.Assert(argument.Kind != BoundKind.DiscardedExpression);
```

Comment:
>DiscardedExpression [](start = 56, length = 19)

But it *could* be a discarded expression, couldn't it? It just should, at this point, be one with a type.